text/0055-remove-unspecified.md

```diff
@@ -0,0 +1,194 @@
+
+The current implementation of unspecified entities results in several corner cases in our code (especially the validator), which have been difficult to maintain (see discussion on [this recent PR](https://github.com/cedar-policy/cedar/pull/603)). The concept of unspecified entities is (in our opinion) confusing for users, and will only become more confusing once we stabilize the partial evaluation experimental feature, which allows “unknown” entities that are subtly different from “unspecified” entities.
+
+This RFC proposes to drop Cedar support for unspecified entities, instead recommending that users create their own application-specific entity types that act like unspecified entities.
+
+## Basic example
+
+Cedar uses a Principal, Action, Resource, Context \<P,A,R,C\> model for requests.
+This model can feel awkward in cases where there is not a clear principal and/or resource for an action.
+For example, consider a "getFile" action.
+The principal here should be a `User`, and the context can store information about the file to get, but it is unclear what the resource should be.
+In this case, a user may want to make the authorization request with a “dummy” resource.
+This is what Cedar’s unspecified entities are for.
+
+As of Cedar v3.1, the only way to “construct” an unspecified entity is by passing `None` as a component when building a [`Request`](https://docs.rs/cedar-policy/3.0.1/cedar_policy/struct.Request.html).
+So a request for this action might look like \<`Some(User::"alice")`, `Some(Action::"getFile")`, `None`, ...\> (using pseudo-syntax).
+
+In the schema, omitting the relevant `appliesTo` field says that an action applies to an unspecified entity.
+For example, the following schemas specify that the "getFile" action applies (only) to an unspecified resource.
+
+Cedar schema syntax ([RFC 24](https://github.com/cedar-policy/rfcs/blob/main/text/0024-schema-syntax.md)):
+
+```
+action getFile
+  appliesTo { principal: [User] };
+```
+
+Cedar schema JSON syntax:
+
+```json
+"getFile": {
+    "appliesTo": {
+        "principal": [User]
+    }
+}
+```
+
+Now consider a "createAccount" action.
+It's unclear what principal _or_ resource should be used with this action.
+The approach above can be extended to this case too: make the action apply to an unspecified principal _and_ resource.
+But now "unspecified" entities are being used to represent multiple distinct concepts: a file store, (unidentified) users creating an account, and a system that manages accounts.
+
+This RFC proposes that users should instead explicitly create these "dummy" entities in an application-specific manner.
+For this example, the user should create entities of three types: `FileSystem`, `UnauthenticatedUser`, and `AccountManager`.
+The first will act as the principal for a "getFile" request, and the second and third will act as the principal and resource for a "createAccount" request.
```

Comment:
as the _resource_ for a "getFile" request?

Based on the discussion it seems like the action is more of create-file than get-file? or maybe it's more like list-files?